src/App/Fossa/Container/Sources/JarAnalysis.hs

```diff
@@ -0,0 +1,52 @@
+{-# LANGUAGE DerivingVia #-}
+
+module App.Fossa.Container.Sources.JarAnalysis (
+  analyzeContainerJars,
+) where
+
+import App.Fossa.EmbeddedBinary (BinaryPaths, toPath, withMillhoneBinary)
+import Container.Types (DiscoveredJars)
+import Control.Algebra (Has)
+import Control.Effect.Diagnostics (Diagnostics, context, fatal)
+import Control.Effect.Lift (Lift)
+import Control.Monad (unless)
+import Data.Aeson (
+  eitherDecode,
+ )
+import Data.ByteString.Lazy qualified as BL
+import Data.String.Conversion (decodeUtf8, toText)
+import Data.Text (Text)
+import Effect.Exec (AllowErr (Never), CmdSuccess (..), Command (..), Exec, execReturningStderr)
+import Effect.Logger (Logger, logDebug, pretty)
+import Path (Abs, File, Path, parent)
+
+-- | Generate a command to run millhone's jar analyze code given a millhone binary and tar file path.
+millhoneJarAnalyzeCmd :: BinaryPaths -> Path Abs File -> Command
+millhoneJarAnalyzeCmd cmdPath imageTarFile =
+  Command
+    { cmdName = toText . toPath $ cmdPath
+    , cmdArgs =
+        [ "--log-to"
+        , "stderr"
+        , "analyze-jars"
+        , toText imageTarFile
+        ]
+    , cmdAllowErr = Never
+    }
+
+-- | Analyze a single Jar file for fingerprints using Millhone.
```

Comment:
```suggestion
-- | Analyze a container for Jar fingerprints using Millhone.
```

Reply:
Good catch, I've changed this along with some of the other work.